components/collapsible-panel/collapsible-panel.js

```diff
@@ -0,0 +1,405 @@
+					75% { transform: rotate(-3deg); animation-timing-function: ease-in-out; }
+					100% { transform: rotate(0deg); animation-timing-function: ease-in-out; }
+				}
+				/* stylelint-enable */
+			}
+			.d2l-collapsible-panel-divider {
+				border-bottom: 1px solid var(--d2l-color-mica);
+				margin-inline: var(--d2l-collapsible-panel-spacing-inline);
+				opacity: 1;
+			}
+			:host([type=inline]) .d2l-collapsible-panel-divider {
+				opacity: 0;
+			}
+			:host(:not([expanded])) .d2l-collapsible-panel:not(.has-summary) .d2l-collapsible-panel-divider {
+				opacity: 0;
+			}
+			.d2l-collapsible-panel-summary,
+			.d2l-collapsible-panel-content {
+				padding: 0.9rem var(--d2l-collapsible-panel-spacing-inline);
+			}
+			:host([type=inline]) .d2l-collapsible-panel-summary,
+			:host([type=inline]) .d2l-collapsible-panel-content {
+				padding-top: 0;
+			}
+			:host([type="inline"]) .d2l-collapsible-panel-summary {
+				margin-top: -0.3rem;
+			}
+			.d2l-collapsible-panel:not(.has-summary) .d2l-collapsible-panel-summary {
+				display: none;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.expanded = false;
+		this.headingLevel = defaultHeading;
+		this.headingStyle = defaultHeading;
+		this.padding = 'default';
+		this.type = 'default';
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+
+		if (!this.expandCollapseLabel) {
+			this.expandCollapseLabel = this.title;
+		}
+
+		if (this.headingStyle === defaultHeading && this.headingLevel !== this.headingStyle) {
+			this.headingStyle = this.headingLevel;
+		}
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._intersectionObserver) this._intersectionObserver.disconnect();
+	}
+
+	firstUpdated() {
+		if (typeof(IntersectionObserver) === 'function') {
+			this._intersectionObserver = new IntersectionObserver(entries => {
+				if (!this.expanded) return;
+
+				const entry = entries[0];
+				const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+				if (entry.isIntersecting) {
+					element.classList.remove('scrolled');
+				} else {
+					element.classList.add('scrolled');
+				}
+			});
+			this._intersectionObserver.observe(this.shadowRoot.querySelector('.d2l-collapsible-panel-top-sentinel'));
+		}
+	}
+
+	render() {
+		return html`
+			<button
+				aria-expanded="${this.expanded}"
+				class="d2l-offscreen"
+				type="button"
+				@click="${this._toggleExpand}"
+				@focus="${this._onFocus}"
+				@blur="${this._onBlur}"
+			>${this.expandCollapseLabel}</button>
+			<div class="d2l-collapsible-panel" @click="${this._handlePanelClick}">
+				<div class="d2l-collapsible-panel-top-sentinel"></div>
+				${this._renderHeader()}
+				<d2l-expand-collapse-content ?expanded="${this.expanded}">
+					<div class="d2l-collapsible-panel-content">
+						<slot name="content"></slot>
+					</div>
+				</d2l-expand-collapse-content>
+				<d2l-expand-collapse-content ?expanded="${!this.expanded}">
```

Comment:
Could this entire part not be rendered if there's no summary elements in the slot? You may need to make `this.hasSummary` into a state property instead of just a CSS class.

Reply:
Yeah, I initially had `_hasSummary` as a state property, but that meant I couldn't use it in css selectors.

That said, the component has changed a lot since then...maybe it's possible to not need `has-summary` in css 🤔

Reply:
Hmmm, so looked into this a bit.

I'm not entirely sure it's possible.... `hasSummary` is set on the `@slotchange` event of `summary`. Which, if we don't render it, will never fire (?) 🤔

Reply:
Hahaha right. Maybe it could at least get tied to a `hidden` on the `d2l-expand-collapse-content`? That would at least ensure it never takes up any space.

Reply:
I tried this -- adding `hidden` alongside `expanded` creates weird issues. Expand-collapse-content does handle this for us though:

https://github.com/BrightspaceUI/core/blob/c72043d2b75c01a7fca6f491d168949f5d92688c/components/expand-collapse/expand-collapse-content.js#L56-L58

Reply:
Ok cool!